data/json/terrain.json

```diff
@@ -7704,6 +8008,69 @@
     "flags": [ "TRANSPARENT", "FLAT" ],
     "bash": { "str_min": 30, "str_max": 210, "sound": "crash!", "sound_fail": "whump!", "ter_set": "t_null", "bash_below": true }
   },
+  {
+    "type": "terrain",
+    "id": "t_tar_flat_roof",
+    "name": "tar paper flat roof",
+    "looks_like": "t_flat_roof",
+    "symbol": ".",
+    "color": "gray",
+    "move_cost": 2,
+    "flags": [ "TRANSPARENT", "FLAMMABLE", "FLAT" ],
+    "bash": { "str_min": 30, "str_max": 210, "sound": "crash!", "sound_fail": "whump!", "ter_set": "t_null", "bash_below": true }
```

Comment:
There's enough going on to skip it in this PR but at some point we should add tar paper as an item, and let you retrieve it from roofs and use it to build roofs.

Reply:
the original t_flat_roof isn't constructable (or deconstructable).  I think it is due to the floor/roof tiles.  I'd rather keep these true to the original unless we decide all flat_roof tiles should be treated this way.  Maybe once that floor/roof tile becomes something like floor/rafters (similar to what you mention below).  Happy to do this but I would rather make it a new PR.